hop/auth.py

```diff
@@ -5,26 +5,48 @@
 
 import argparse
 import errno
+import getpass
 import logging
 import os
 
+import toml
+
 from adc import auth
 
 DEFAULT_AUTH_CONFIG = """\
-security.protocol=SASL_SSL
-sasl.mechanism=SCRAM-SHA-512
-sasl.username=username
-sasl.password=password
+[auth]
+username = "{username}"
+password = "{password}"
 """
 
 logger = logging.getLogger("hop")
 
-# expose auth options from adc
-SASLAuth = auth.SASLAuth
 SASLMethod = auth.SASLMethod
 
-# map default auth option to Auth
-Auth = auth.SASLAuth
+
+class Auth(auth.SASLAuth):
+    """Attach SASL-based authentication to a client.
+
+    Returns client-based auth options when called.
+
+    Parameters
+    ----------
+    user : `str`
+        Username to authenticate with.
+    password : `str`
+        Password to authenticate with.
+    ssl : `bool`, optional
+        Whether to enable SSL (enabled by default).
+    method : `SASLMethod`, optional
+        The SASL method to authenticate, default = SASLMethod.SCRAM_SHA_512.
+        See valid SASL methods in SASLMethod.
+    ssl_ca_location : `str`, optional
+        If using SSL via a self-signed cert, a path/location
+        to the certificate.
+    """
+
+    def __init__(self, user, password, ssl=True, method=SASLMethod.SCRAM_SHA_512, **kwargs):
```

Comment:
Why `**kwargs` instead of explicitly including `ssl_ca_location`?

Reply:
No good reason, just didn't want to unintentionally miss any options.